central/config/datastore/singleton.go

```diff
@@ -149,6 +160,10 @@ func validateConfigAndPopulateMissingDefaults(datastore DataStore) {
 		needsUpsert = true
 	}
 
+	if features.CustomizablePlatformComponents.Enabled() {
+		needsUpsert = populateDefaultSystemRuleIfMissing(config)
+	}
+
```

Comment:
```suggestion
	if features.CustomizablePlatformComponents.Enabled() && populateDefaultSystemRuleIfMissing(config) {
		needsUpsert = true
	}

```

I think this is incorrect.  We could set `needsUpsert` to `false` even if it was set to `true` above.

Reply:
Right. Good catch